airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/components.py

```diff
@@ -63,7 +63,13 @@ class KeyValueExtractor(RecordExtractor):
     values_extractor: RecordExtractor
 
     def extract_records(self, response: requests.Response) -> Iterable[MutableMapping[Any, Any]]:
-        keys = self.keys_extractor.extract_records(response)
+        keys = list(self.keys_extractor.extract_records(response))
         values = self.values_extractor.extract_records(response)
 
-        yield dict(zip(keys, values))
+        has_more_chunks = True
```

Comment:
i noticed that we're switching this back to the original logic from the python version of the integration. I'm fine with that since it seems like we do this chunking for a reason so my initial suggestion to simplify probably missed something.

But in the above docstring, can you show an example for multiple records on each extractor to demonstate how the zip/chunking logic works. I think that will be helpful since it does seem like the `keys_extractor` and `values_extractor` don't have exactly the same number of records and it would be good to understand how this works with a more complex example

Reply:
We start with a list of field names (keys), for example:
`['date', 'firstUserCampaignName', 'firstUserMedium']`.

Then we receive a flat sequence of values in that exact order (values), for example:
`['20231001', 'TikTok-Conversion', 'ads', '20231001', 'TikTok-Mahta', 'ads']`.

Because there are three fields, we split the values into chunks of three. Each chunk maps onto the keys to form one record:

```python
 {
  'date': '20231001',
   'firstUserCampaignName': 'TikTok-Conversion',
   'firstUserMedium': 'ads'
 }

 {
   'date': '20231001',
   'firstUserCampaignName': 'TikTok-Mahta',
   'firstUserMedium': 'ads'
 }
```